tf/edgeml/trainer/protoNNTrainer.py

```diff
@@ -0,0 +1,152 @@
+import tensorflow as tf
+import numpy as np
+
+
+class ProtoNNTrainer:
+    def __init__(self, protoNNObj, regW, regB, regZ,
+                 sparcityW, sparcityB, sparcityZ,
+                 learningRate, X, Y):
+        '''
+        protoNNObj: An instance of ProtoNN class. This instance
+            will be trained.
+        regW, regB, regZ: Regularization constants for W, B, and
+            Z matrices of protoNN.
+        sparcityW, sparcityB, sparcityZ: Sparcity constraints
+            for W, B and Z matrices.
+        X, Y : Placeholders for data and labels.
+            X [-1, featureDimension]
+            Y [-1, num Labels]
+        '''
+        self.protoNNObj = protoNNObj
+        self.__regW = regW
+        self.__regB = regB
+        self.__regZ = regZ
+        self.__sW = sparcityW
+        self.__sB = sparcityB
+        self.__sZ = sparcityZ
+        self.__lR = learningRate
```

Comment:
All the reg, spar and learning rate should be public, THis will help the users change them before re-running the same trainer again (retrain from the old solution)

Reply:
Good point. I'll have `getSparcity`, `setSparcity` methods. Directly writing to variables is a no no.